c7n/resources/redshift.py

```diff
@@ -799,3 +830,53 @@ def process_resource_set(self, resources, tag_keys):
             arn = self.manager.generate_arn(
                 r['ClusterIdentifier'] + '/' + r['SnapshotIdentifier'])
             client.delete_tags(ResourceName=arn, TagKeys=tag_keys)
+
+
+@RedshiftSnapshot.action_registry.register('revoke-access')
+class RedshiftSnapshotRevokeAccess(BaseAction):
+    """Revokes ability of accounts to restore a snapshot
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: redshift-snapshot-revoke-access
+                resource: redshift-snapshot
+                filters:
+                  - type: cross-account
+                    whitelist:
+                      - 012345678910
+                actions:
+                  - type: revoke-access
+    """
+    permissions = ('redshift:RevokeSnapshotAccess',)
+    schema = type_schema('revoke-access')
+
```

Comment:
the action should probably have validate method that verifies its being used in conjunction with a cross-account filter in the policy, else its a no-op.

Reply:
@kapilt wouldn't this be implicit due to the check against `'c7n:CrossAccountViolations'` associated to each snapshot?